rfcs/convergence/slot-null-rendering.md

```diff
@@ -0,0 +1,232 @@
+
+```tsx
+// react-utilities/src/compose/getSlots.ts
+// ....
+if (typeof slot === 'string' && children === undefined) {
+  slot = nullRender;
+}
+```
+
+## Problem statement
+
+The verification of only allowing native elements to be rendered if this element present `children` as property breaks in some edge cases
+
+### [Empty Elements]
+
+The `Input` component is a good case example of that as [reported](https://github.com/microsoft/fluentui/pull/18642#issuecomment-871677964) by @ecraig12345, since the native element `input` is an [Empty Element], it cannot have any children, and still needs to be rendered.
+
+Right now for input case this is the solution to avoid `children = undefined` problem:
+
+```tsx
+export function Input(props) {
+  const state = {
+    components: {
+      input: 'input'
+    },
+    input: resolveShorthand(props.input, {
+      children: React.Fragment // 🚨 getSlots requires children
+    })
+  };
+  const { slots, slotProps } = getSlots(state, ['input']);
+  delete slotProps.input.children; // 🚨 input can't have children
+
+  return (
+    <slots.root {...slotProps.root}>
+      <slots.input {...slotProps.input} />
+    </slots.root>
+  );
+}
+```
+[Live Example](https://stackblitz.com/edit/react-ts-wwguyp?file=Input.tsx)
+
+
+### Slot as a parent for other slots
+
+This edge case has been seen in different converged components ([`CompoundButton`](https://github.com/microsoft/fluentui/blob/master/packages/react-button/src/components/CompoundButton/renderCompoundButton.tsx#L18), [`AccordionHeader`](https://github.com/microsoft/fluentui/blob/master/packages/react-accordion/src/components/AccordionHeader/renderAccordionHeader.tsx#L15))
+
+The problem is that using a slot as parent for other slots means overriding `children` props, so `children` is not declared in the moment `resolveShorthand` is invoked but in the moment that the slot will be rendered.
+
+
+```tsx
+export function Component(props) {
+  const state = {
+    components: {
+      button: 'button',
+      icon: 'i'
+    },
+    button: resolveShorthand(props.button, {
+      children: React.Fragment // 🚨 getSlots requires children
+    }),
+    icon: resolveShorthand(props.icon)
+  };
+
+  const { slots, slotProps } = getSlots(state, ['input']);
+
+  return (
+    <slots.root {...slotProps.root}>
+      <slots.button {...slotProps.button}>
+        <slots.icon {...slotProps.icon} />
+      </slots.button>
+    </slots.root>
+  );
+}
+```
+[Live Example](https://stackblitz.com/edit/react-ts-wwguyp?file=SlotAsParent.tsx)
+
+### Native elements without children
+
+e.g: An Icon that doesn't have children `<i class="icon"/>` (this cases is hypothetical, but quite possible, as we cannot verify it in converged components)
+
+## Detailed Design or Proposal
+
+Both options here proposed by the end are equivalent, the only difference is the impact on `ObjectShorthandProps` interface
+
+### Option 1: Make `shorthands` optional
```

Comment:
This might also be more obvious to people familiar with the helpers, but where is this `shorthands` defined?

Reply:
It is actually defined in the [Background](https://github.com/microsoft/fluentui/blob/8eb711e7e07020e178ba1e62722326450433cc3a/rfcs/convergence/slot-null-rendering.md#background)

> getSlots is a method that iterates over ObjectShorthandProps (shorthands) declarations and converts them to something that can be rendered by React.

I'll try to make it more explicit!